src/relion/zocalo/wrapper.py

```diff
@@ -143,12 +143,18 @@ def run(self):
                 logger.info("Sent %d commands to ISPyB", len(ispyb_command_list))
 
             # if Relion has been running too long stop loop of preprocessing jobs
-            most_recent_movie = max(
-                [
-                    p.stat().st_mtime
-                    for p in pathlib.Path(self.params["image_directory"]).glob("**/*")
-                ]
-            )
+            try:
+                most_recent_movie = max(
+                    [
+                        p.stat().st_mtime
+                        for p in pathlib.Path(self.params["image_directory"]).glob(
+                            "**/*"
+                        )
+                    ]
+                )
```

Comment:
`max()` can consume an iterable directly, so you don't need to create the temporary list
```suggestion
                most_recent_movie = max(
                    p.stat().st_mtime
                    for p in pathlib.Path(self.params["image_directory"]).glob(
                        "**/*"
                    )
                )
```

Reply:
On second thought, you could probably get rid of the race condition entirely using [`os.scandir`](https://docs.python.org/3/library/os.html#os.scandir).
```python
with os.scandir(self.params["image_directory"]) as dir:
    most_recent_movie = max(e.stat().st_mtime for e in dir)
```
which would, as a side-effect, give better performance with many files - especially over the network. (`scandir` lists the directory and gets the `stat()` information at the same time, so you have 1 operation instead of 1+n operations for n files)

However, this is not recursive. You could recurse using `e.is_dir()`, but that'd be a bit more work.

Reply:
I'll take that back. `.stat()` on a scandir `DirEntry` object will still cause an `lstat()` call. So it would neither fix the race condition nor improve performance